pycontrails/models/apcemm/adapter.py

```diff
@@ -0,0 +1,831 @@
+            ) as f:
+                status = f.read().strip()
+                statuses.append(status)
+
+            # Get segment initialization time, estimated as
+            # average of waypoint times at segment start and end
+            base_time = self.source.dataframe.iloc[segment : segment + 1]["time"].mean()
+
+            # Convert contents of wake vortex output to pandas dataframe
+            # with elapsed times converted to absolute times
+            vortex = pd.read_csv(
+                self.apcemm_file(segment, os.path.join(output_directory, "Micro000000.out")),
+                skiprows=[1],
+            ).rename(columns=lambda x: x.strip())
+            time = (base_time + pd.to_timedelta(vortex["Time [s]"], unit="s")).rename("time")
+            waypoint = pd.Series(np.full((len(vortex),), segment), name="waypoint")
+            vortex = pd.concat((waypoint, time, vortex.drop(columns="Time [s]")), axis="columns")
+            vortexes.append(vortex)
+
+            # Record paths to contrail output (netCDF files) in pandas dataframe
+            # get paths to contrail output
+            files = sorted(
+                glob.glob(
+                    self.apcemm_file(
+                        segment, os.path.join(output_directory, "ts_aerosol_case0_*.nc")
+                    )
+                )
+            )
+            if len(files) == 0:
+                continue
+            time = []
+            path = []
+            for file in files:
+                elapsed_hours = pd.to_timedelta(file[-7:-5] + "h")
+                elapsed_minutes = pd.to_timedelta(file[-5:-3] + "m")
+                elapsed_time = elapsed_hours + elapsed_minutes
+                time.append(base_time + elapsed_time)
+                path.append(file)
+            waypoint = pd.Series(np.full((len(time),), segment), name="waypoint")
+            contrail = pd.DataFrame.from_dict(
+                {
+                    "waypoint": waypoint,
+                    "time": time,
+                    "path": path,
+                }
+            )
+            contrails.append(contrail)
+
+        # Mark status of final waypoint as n/a (no segment)
+        statuses.append("N/A")
+
+        # Attach status to self
+        self.source["status"] = statuses
+
+        # Attach wake vortex and contrail outputs to model
+        self.vortex = pd.concat(vortexes, axis="index", ignore_index=True)
+        if len(contrails) > 0:  # only present if APCEMM simulates persistent contrails
+            self.contrail = pd.concat(contrails, axis="index", ignore_index=True)
+
+    @property
+    def dynamic_yaml_params(self) -> list[str]:
+        """List of :class:`APCEMMYaml` attributes set dynamically by this model.
+
+        Other :class:`APCEMMYaml` attributes can be set statically by passing a custom instance of
+        :class:`APCEMMYaml` when the model is created.
+        """
+        return [
```

Comment:
Can this somehow be introspected from the `APCEMMInput` fields?  e.g.

```
yaml = APCEMMInput()
dataclasses.asdict(yaml).keys()
```

Reply:
Nothing comes to mind for me.... I did clean up the way APCEMM input parameters are set in 1dc2173b5f20445833c83ac16d15a3571d3001d9 (in particular, users now get an error early on if they try to override a parameter that the interface sets), so hopefully that helps make it clear who is responsible for which inputs.